core/qbft/qbft.go

```diff
@@ -451,6 +442,24 @@ func nextMinRound[I any, V comparable](d Definition[I, V], frc []Msg[I, V], roun
 	return rmin
 }
 
+// isJustified returns true if message is justified or if it does not need justification
+func isJustified[I any, V comparable](d Definition[I, V], instance I, msg Msg[I, V]) bool {
+	switch msg.Type() {
+	case MsgPrePrepare:
+		return isJustifiedPrePrepare(d, instance, msg)
+	case MsgPrepare:
+		return true
+	case MsgCommit:
+		return true
+	case MsgRoundChange:
+		return isJustifiedRoundChange(d, msg)
+	case MsgDecided:
+		return isJustifiedDecided(d, msg)
+	default:
+		return false
```

Comment:
So I have been using panics as sanity checks in the algo, since this case is unexpected, I would panic here: `panic("bug: invalid message type")`

Reply:
@corverroos I could add a panic, but I would consider it tech debt, we should handle all errors.

Reply:
Yeah I know people avoid panics, so I am breaking the "rules". But I opted for panics for sanity checks to simplify the code and to distinguish between normal errors (like when doing networking) and sanity checks (things that fatal bugs that should never happen). It also makes the code more explicit, stating what is assumptions about the state of the world.

Reply:
Done